pkg/controllers/deprovisioning/controller.go

```diff
@@ -168,134 +153,57 @@ func (c *Controller) deprovision(ctx context.Context, deprovisioner Deprovisione
 	return true, nil
 }
 
-func (c *Controller) executeCommand(ctx context.Context, d Deprovisioner, command Command) error {
+func (c *Controller) executeCommand(ctx context.Context, d Deprovisioner, cmd Command) error {
```

Comment:
question: Does it make sense to throw the entire `executeCommand` inside of the queuing mechanism and let the `Add` command handle the synchronous bits so that all of the business logic is in one place?

Reply:
Good thought. I kinda like keeping these separate here. If any line fails in the executeCommand, there'd be no reason to requeue/try again, as we just need to find another command. From the perspective of a replace action, if the initial launches fail, there's nothing to wait for, which is what the main value of the queue is.